### PR TITLE
Make ParallelInfo executable display info about parallelism

### PR DESCRIPTION
**Pull request by nilsdeppe**

### CODE REVIEW

File: src/Executables/ParallelInfo/ParallelInfo.cpp [hunk 1]
```diff
@@ -1,13 +1,90 @@
...
+#include "Parallel/Info.hpp"
+#include "Parallel/Printf.hpp"
 
-extern "C" void CkRegisterMainModule(void) {}
+/// \cond HIDDEN_SYMBOLS
+ParallelInfo::ParallelInfo(CkArgMsg* msg) {
+  Parallel::printf("Executing '%s' using %d processors.\n", msg->argv[0],
+                   Parallel::number_of_procs());
+  if (msg->argc > 1) {
+    ERROR("Expected zero messages, not " << msg->argc - 1);
```
Comment (kidder) [inline src/Executables/ParallelInfo/ParallelInfo.cpp]: messages -> command line options
and print the ones received

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
Also, I think it might be clearer to define callbacks instead of counting; thoughts?

**Comment by nilsdeppe** [comment]
I agree, I'll make the change to use a reduction with a callback.

File: src/Executables/ParallelInfo/ParallelInfo.cpp [hunk 2]
```diff
@@ -34,49 +40,64 @@ ParallelInfo::ParallelInfo(CkArgMsg* msg) {
...
+    if (number < 0) {
+      digits = 1;
+    }
+    while (number) {
+      number /= 10;
+      digits++;
+    }
+    return digits;
+  };
+  const int digits_in_node = compute_digits(Parallel::number_of_nodes() - 1);
```
Comment (kidder) [inline src/Executables/ParallelInfo/ParallelInfo.cpp]: how about using log10

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Updated, squashed commits since there was only the one change.

File: src/Executables/ParallelInfo/ParallelInfo.cpp [hunk 3]
```diff
@@ -1,13 +1,105 @@
...
+  Parallel::printf("\nReport finished.\n");
+  Parallel::exit();
 }
+
+// Helper print function so that both node and pe group prints can be changed
+// easily.
+void print_info() {
+  auto compute_digits = [](const auto number) {
+    return number == 0 ? 1 : static_cast<decltype(number)>(
+                                 std::ceil(std::log10(std::abs(number))));
```
Comment (wthrowe) [inline src/Executables/ParallelInfo/ParallelInfo.cpp]: Incorrect for powers of 10.  Should be `std::ceil(std::log10(std::abs(number) + 1))`.

**Comment by nilsdeppe** [comment]
Good catch! I had talked to Larry about adding a non-constexpr math header file, since some of these functions we will want in general. So far the best we could come up with is `Utilities/Math.hpp`, but if you have a better suggestion I'd be happy to move it. I'm not sure if we'll want one test per math function, one for all of them, or some middle ground. This will likely be answered as we add more. I was thinking that these remain basic math functions that only depend on `<cmath>` and type traits, at least until we find a need for more complicated things.

**Review by kidder — APPROVED** [review]

**Pull request merged**